### PR TITLE
Add countdown style current time manipulations; Edit Douban.fm connector; Add Douban Artists connector

### PR DESCRIPTION
**Pull request by xyauhideto**
The `currentTime` of the two connectors depends on this manipulation.

### CODE REVIEW

File: connectors/douban.fm.js [hunk 1]
```diff
@@ -1,22 +1,14 @@
...
-
-Connector.getDuration = () => {
-	let text = $('.time').text();
-	return Util.stringToSeconds(text.substring(1));
-};
-
-Connector.isPlaying = () => {
-	return $('.player-wrapper .buttons .icon-pause').length > 0;
-};
+const update = (attrs) => {
```
Comment (alexesprit) [inline connectors/douban.fm.js]: Is it a certain purpose to add this function?

Comment (xyauhideto) [inline connectors/douban.fm.js]: Just for brevity. If it violates something, I will change it.

Comment (xyauhideto) [reply connectors/douban.fm.js]: Just for brevity. If it violates something, I will change it.

Comment (alexesprit) [reply connectors/douban.fm.js]: I would keep all connectors written in the same style (the one rest of all are written in).

File: connectors/douban.fm.js [hunk 2]
```diff
@@ -1,22 +1,14 @@
...
+		Connector[key] = attrs[key];
+	}
+};
+update({
+	playerSelector: '.player-wrapper',
+	trackArtSelector: '.playing-cover img',
+	trackSelector: '.middle>div:nth-child(2)>a',
+	artistSelector: '.middle>div:nth-child(3) a',
+	playButtonSelector: '.icon-play',
+	getCurrentTime: () => -Util.stringToSeconds($('.time').text().substring(1)),
```
Comment (alexesprit) [inline connectors/douban.fm.js]: The controller now receives negative duration.

Comment (xyauhideto) [reply connectors/douban.fm.js]: That's the played time of the counting down style

File: connectors/douban.fm.js [hunk 3]
```diff
@@ -1,22 +1,14 @@
...
+	for (let key in attrs) {
+		Connector[key] = attrs[key];
+	}
+};
+update({
+	playerSelector: '.player-wrapper',
+	trackArtSelector: '.playing-cover img',
+	trackSelector: '.middle>div:nth-child(2)>a',
+	artistSelector: '.middle>div:nth-child(3) a',
+	playButtonSelector: '.icon-play',
```
Comment (alexesprit) [inline connectors/douban.fm.js]: Play/pause doesn't work for me. Have you the same issue?

Btw, I fixed it for myself by adding `isPlaying` function:
```js
Connector.isPlaying = () => $('.icon-pause').length > 0;
```

File: connectors/douban.fm.js (hunk 2 above)
Comment (xyauhideto) [reply connectors/douban.fm.js]: The former one assigns the absolute value to the duration, continuously triggering onStatusChanged events.

Comment (alexesprit) [reply connectors/douban.fm.js]: Yes, that was a bug.

File: connectors/douban.fm.js (hunk 3 above)
Comment (xyauhideto) [reply connectors/douban.fm.js]: I'm not sure what's expected.
No test report was triggered and the record was scrobbled after pausing in both cases.

Comment (alexesprit) [reply connectors/douban.fm.js]: The extension should not scrobble paused tracks.

Comment (xyauhideto) [reply connectors/douban.fm.js]: Still the track was scrobbled, ten minutes after first detection.

File: core/content/connector.js [hunk 4]
```diff
@@ -185,7 +185,12 @@ var BaseConnector = window.BaseConnector || function() {
 	 * @return {Number} Number of seconds passed from the beginning of the track
 	 */
 	this.getCurrentTime = function() {
-		let text = $(this.currentTimeSelector).text();
+		let text = $(this.currentTimeSelector).text(),
+			isCountDown = text.indexOf('-') + 1;
+		if (isCountDown) {
+			text = text.substr(isCountDown);
+			return (this.duration || 0) - Util.stringToSeconds(text);
```
Comment (alexesprit) [inline core/content/connector.js]: Is there a `duration` property of `Connector` object?

**Comment by alexesprit** [comment]
Please next time use several PRs for each change (translations, connector improvements etc). It's really hard to review all these changes in one PR.

File: core/content/connector.js (hunk 4 above)
Comment (xyauhideto) [reply core/content/connector.js]: I see. It should be `getDuration()`.

**Comment by xyauhideto** [comment]
Sorry for put everything in single PR as this is the first time I use git and alike for collaboration.
Now I've separated them #1428 #1429 #1430 #1431 #1432 , hope I'm doing this right.

**Pull request merged**